iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/DetachLineFromLines.java

```diff
@@ -0,0 +1,229 @@
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.NetworkModification;
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.Line;
+import com.powsybl.iidm.network.LineAdder;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.VoltageLevel;
+
+import java.util.Objects;
+
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.LoadingLimitsBags;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.addLoadingLimits;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.attachLine;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.calcMinLoadingLimitsBags;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.createLineAdder;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.createdLineReport;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.noAttachmentPointAndOrAttachedVoltageLevelReport;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.notFoundLineReport;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.removeVoltageLevelAndSubstation;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.removedLineReport;
+
+/**
+ * This method reverses the action done in the AttachNewLineOnLine class :
+ * it replaces 3 existing lines (with the same voltage level at one of their side) with a new line,
+ * and eventually removes the existing voltage levels, if they contain no equipments anymore, except bus or bus bar section
+ *
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class DetachLineFromLines implements NetworkModification {
+
+    private String lineAZId;
+    private String lineBZId;
+    private String lineCZId;
+
+    private String lineId;
+    private String lineName;
+
+    private static final String LINE_NOT_FOUND_REPORT_MESSAGE = "Line %s is not found";
+
+    /**
+     * Constructor.
+     *
+     * @param lineAZId     The non-null ID of the first line
+     * @param lineBZId     The non-null ID of the second line
+     * @param lineCZId     The non-null ID of the third line (connecting attachment point to attached voltage level)
+     * @param lineId       The non-null ID of the new line to be created
+     * @param lineName     The optional name of the new line to be created
+     */
+    public DetachLineFromLines(String lineAZId, String lineBZId, String lineCZId, String lineId, String lineName) {
+        this.lineAZId = Objects.requireNonNull(lineAZId);
+        this.lineBZId = Objects.requireNonNull(lineBZId);
+        this.lineCZId = Objects.requireNonNull(lineCZId);
+        this.lineId = Objects.requireNonNull(lineId);
+        this.lineName = lineName;
+    }
+
+    public DetachLineFromLines setLineAZId(String lineAZId) {
```

Comment:
Why getters ans setters ?

Reply:
They are tested in testConstructor and testSetters tests